common/json.c

```diff
@@ -838,7 +838,12 @@ void json_add_tok(struct json_stream *result, const char *fieldname,
 		if (json_tok_is_num(buffer, tok)) {
 			json_to_int(buffer, tok, &i);
 			json_add_num(result, fieldname, i);
-		}
+		} else if (buffer[tok->start] == 't' || buffer[tok->start] == 'T')
```

Comment:
Garbage in, garbage out.

If they feed in an invalid primitive, we should either pass it through, or get upset with the passer.

But I can't understand why json_add_tok exists.  Why are we trying to normalize tokens?  We should just append them as we're asked to.

@darosior ?

Reply:
> But I can't understand why json_add_tok exists. Why are we trying to normalize tokens? We should just append them as we're asked to.

Using `json_add_member` ? I don't recall exactly but it seems cleaner to use our already-present helpers.

Anyway, i 1) don't know how comes i only handled the num primitive and 2) am pretty sure `Null` `False` and `True` here and below are invalid JSON (maybe just use `json_add_literal` in the `else` branch).

Reply:
It also turns strings into bools, etc.  It's just *weird*.

I'm testing the simplest variant now:
```C
void json_add_tok(struct json_stream *result, const char *fieldname,
                  const jsmntok_t *tok, const char *buffer)
{
	char *space;
	assert(tok->type != JSMN_UNDEFINED);

	space = json_member_direct(result, fieldname, tok->end - tok->start);
	memcpy(space, buffer + tok->start, tok->end - tok->start);
}
```